### PR TITLE
Card: Add ApplicationContext to ConfirmPaymentSourceRequest

### PR DESCRIPTION
**Pull request by jcnoriega**
### Reason for changes

There was a bug in `ordersv2` where we had to add `appliction_context` to the order creation request, in order for 3DS to work. Now, we only have to add it in the `confirm_payment_source` request, removing it from order creation

Jira Ticket: https://engineering.paypalcorp.com/jira/browse/DTNOR-606

### Summary of changes

- Add Application context to ConfirmPaymentSourceRequestFactory

### Checklist

- [ ] Added a changelog entry

- @jcnoriega

### CODE REVIEW

File: Demo/Demo/ViewModels/BaseViewModel.swift [hunk 1]
```diff
@@ -57,8 +57,7 @@ class BaseViewModel: ObservableObject, PayPalWebCheckoutDelegate, CardDelegate {
         let amountRequest = Amount(currencyCode: "USD", value: amount)
         let orderRequestParams = CreateOrderParams(
             intent: DemoSettings.intent.rawValue.uppercased(),
-            purchaseUnits: [PurchaseUnit(amount: amountRequest)],
-            applicationContext: ApplicationContext(returnUrl: BaseViewModel.returnUrl, cancelUrl: BaseViewModel.cancelUrl)
+            purchaseUnits: [PurchaseUnit(amount: amountRequest)]
```
Comment (jaxdesmarais) [inline Demo/Demo/ViewModels/BaseViewModel.swift:60]: 🥳

File: Sources/Card/ConfirmPaymentSourceRequest.swift [hunk 2]
```diff
@@ -5,50 +5,75 @@ import PaymentsCore
...
     ) throws {
+        var confirmPaymentSourceBody = ConfirmPaymentSourceBody()
         var card = cardRequest.card
         if let threeDSecureRequest = cardRequest.threeDSecureRequest {
             let verification = Verification(method: threeDSecureRequest.sca.rawValue)
             card.attributes = Attributes(verification: verification)
+            
+            let applicationContext = ApplicationContext(
+                returnUrl: threeDSecureRequest.returnUrl,
+                cancelUrl: threeDSecureRequest.cancelUrl
```
Comment (scannillo) [inline Sources/Card/ConfirmPaymentSourceRequest.swift:29]: It looks like we need to update our [GitHub docs ](https://github.com/paypal/iOS-SDK/tree/main/docs/Card#5-create-a-request-containing-the-card-payment-details)to include a code snippet on adding these URLs to the `ThreeDSecureRequest`

File: Sources/Card/ConfirmPaymentSourceRequest.swift [hunk 3]
```diff
@@ -5,50 +5,75 @@ import PaymentsCore
...
-
+    
     var headers: [HTTPHeader: String] {
         [
             .contentType: "application/json", .acceptLanguage: "en_US",
             .authorization: "Bearer \(accessToken)"
         ]
     }
+    
+    private struct ConfirmPaymentSourceBody: Encodable {
```
Comment (jaxdesmarais) [inline Sources/Card/ConfirmPaymentSourceRequest.swift]: nit: are we able to just call this `ConfirmPaymentSource`?

**Review by jaxdesmarais — APPROVED** [review]
👻

File: Sources/Card/ConfirmPaymentSourceRequest.swift (hunk 2 above)
Comment (jcnoriega) [reply Sources/Card/ConfirmPaymentSourceRequest.swift:29]: done!

File: docs/Card/README.md [hunk 4]
```diff
@@ -144,9 +144,15 @@ Optionally, a merchant app can request Strong Consumer Authentication (SCA) for
...
 ```swift
-cardRequest.threeDSecureRequest = ThreeDSecureRequest(sca: .always)
+cardRequest.threeDSecureRequest = ThreeDSecureRequest(
+    sca: .always,
+    returnUrl: "\(Bundle.main.bundleIdentifier)://return_url",
+    cancelUrl: "\(Bundle.main.bundleIdentifier)://cancel_url"
+)
 ```
 
+Where `return_url` and `cancel_url` are the urls configured in your [paypal developer dashboard](https://developer.paypal.com/)
```
Comment (scannillo) [inline docs/Card/README.md:154]: I remember Jon mentioned that this was no longer a requirement, for merchants to need to upload their URLS to the dev dashboard. Do we know which version this requirement was dropped in?

Comment (jcnoriega) [reply docs/Card/README.md:154]: @scannillo i think Jon meant for that for NXO. If i dont put it here, i get an error on the browser.

Comment (scannillo) [reply docs/Card/README.md:154]: Woops, yes that is for NXO!

**Review by scannillo — APPROVED** [review]
Changes look good to me!

I have a question though - is there a benefit to documenting that a merchant use a separate return and cancel URL? This would work fine with 1 return URL (for example `<bundleID>://payments`).

**Comment by sshropshire** [comment]
> I have a question though - is there a benefit to documenting that a merchant use a separate return and cancel URL?

Can we simplify the integration on iOS with `ASWebAuthenticationSession`? I know on BT we removed the requirement for the merchant to provide a custom return url scheme. That could be one less step for merchants unless we want 100% feature parity with Android.

**Comment by jcnoriega** [comment]
@sshropshire the `return_url` and `cancel_url` are a requirment of `ordersV2` api for the 3DS flow right? the `CardClient` already uses a `ASWebAuthenticationSession`. I dont think we can escape the setting of the urls, it might be that we can just ask for `return_url` only, and then set the `cancel_url` on `ApplicationContext` internal ourselves. I think what @scannillo points out is true, we might not need the urls to be diferent

**Comment by sshropshire** [comment]
> @sshropshire the `return_url` and `cancel_url` are a requirment of `ordersV2` api for the 3DS flow right?

Yeah definitely. My question was more of a DX one. I think on Braintree SDK the merchant doesn't need to specify either because of the way iOS works. There's no app switching in iOS, so the return and cancel urls can technically be defined by the SDK internally to make the integration more seamless.

**Comment by jcnoriega** [comment]
@sshropshire good point. I will explore if we can set it within the iOS SDK.

**Comment by jcnoriega** [comment]
@sshropshire @scannillo will create a ticket to follow up with the following: try to set `return_url` and `cancel_url` within the SDK, if not possible, just ask for `return_url`.

**Pull request merged**